python/pyiceberg/typedef.py

```diff
@@ -85,7 +79,7 @@ class StructProtocol(Protocol):  # pragma: no cover
     """A generic protocol used by accessors to get and set at positions of an object"""
 
     @abstractmethod
-    def set_record_schema(self, record_schema: StructType) -> None:
+    def __init__(self, record_schema: Optional[StructType] = None, **data: Any) -> None:
```

Comment:
Do we need this change to `StructProtocol` or can we make the change to `Record` only?

Reply:
Done!